lib/valueflow.cpp

```diff
@@ -4327,14 +4320,9 @@ struct LifetimeStore {
     }
 
 private:
-    Context* mContext{};
-    void forwardLifetime(Token* tok, TokenList& tokenlist, ErrorLogger* errorLogger, const Settings& settings) const {
-        if (mContext) {
-            mContext->tok = tok;
-            mContext->tokenlist = &tokenlist;
-            mContext->errorLogger = errorLogger;
-            mContext->settings = &settings;
-        }
+    mutable Token* forwardTok{};
```

Comment:
I don't know about using the mutable keyword here.

Reply:
Me neither - but removing the `const` from the function had quite a ripple effect before - will check again. I think this might be a sensible use of `mutable`.

Reply:
The ripple effect wasn't as bad as I remember. The object was only `const` when passing it around so that was not fully enforced and is actually fine.